src/router.js

```diff
@@ -29,38 +23,47 @@ export default function (options) {
       }
     ]
   }
-}
 
-function match(routes, path) {
-  var match, params = {}
+  function match(model, data) {
+    var match
+    var params = {}
 
-  for (var route in routes) {
-    var keys = []
+    for (var route in app.view) {
+      var keys = []
 
-    if (route === "*") {
-      continue
-    }
+      if (route !== "*") {
+        data.replace(new RegExp("^" + route
+          .replace(/\//g, "\\/")
+          .replace(/:([A-Za-z0-9_]+)/g, function (_, key) {
+            keys.push(key)
 
-    path.replace(new RegExp("^" + route
-      .replace(/\//g, "\\/")
-      .replace(/:([A-Za-z0-9_]+)/g, function (_, key) {
-        keys.push(key)
-        return "([-A-Za-z0-9_]+)"
-      }) + "/?$", "g"), function () {
+            return "([-A-Za-z0-9_]+)"
+          }) + "/?$", "g"), function () {
 
-        for (var i = 1; i < arguments.length - 2; i++) {
-          params[keys.shift()] = arguments[i]
-        }
-        match = route
-      })
+            for (var i = 1; i < arguments.length - 2; i++) {
+              params[keys.shift()] = arguments[i]
+            }
 
-    if (match) {
-      break
+            match = route
+          })
+      }
+
+      if (match) {
+        break
+      }
     }
-  }
 
-  return {
-    match: match || "*",
-    params: params
+    if (app.hooks && app.hooks.onRoute) {
+      app.hooks.onRoute(model)
+    }
```

Comment:
Thought I had figured it out, but no. 🤔 Tough without rewiring a lot of stuff. Let me see.